app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/analytics/ApkfyManagerProbe.kt

```diff
@@ -18,14 +18,21 @@ class ApkfyManagerProbe(
     private const val GUEST_UID_KEY = "GUEST_UID"
   }
 
-  override suspend fun getApkfy() = apkfyManager.getApkfy()
-    ?.also(::setApkfyUTMProperties)
-    ?.also {
-      idsRepository.saveId(GUEST_UID_KEY, it.guestUid)
-    }.also {
-      // TODO: improve this logic
-      AptoideMMPCampaign.guestUID = it?.guestUid ?: idsRepository.getId(GUEST_UID_KEY)
+  override suspend fun getApkfy(): ApkfyModel? {
+    if (idsRepository.getId(GUEST_UID_KEY).isEmpty()) {
```

Comment:
Once the APKFY will return `null` and return it always, means this call will repeat every app start forever? 🤔